pkg/utils/config.go

```diff
@@ -0,0 +1,89 @@
+package utils
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/fatih/color"
+)
+
+type Config interface {
+	Read() error
+	Write() error
+	Init(rd io.Reader) error
+}
+
+func getConfigDir() (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		log.Printf("%v failed to determine home directory: %v", color.RedString("ERROR:"), err)
```

Comment:
i see a lot of this kind of pattern here, let's modify it so we could just write it like this:
```go
m.Foo("failed to determine home directory: %v", err);
// output -> ERROR: failed to determine home directory: <error>
```

Reply:
so, create a new interface @threeal

Reply:
yup, a new function inside `pkg`.